ampd/src/handlers/mvx_verify_msg.rs

```diff
@@ -0,0 +1,370 @@
+use std::collections::HashSet;
+use std::convert::TryInto;
+
+use async_trait::async_trait;
+use cosmrs::cosmwasm::MsgExecuteContract;
+use cosmrs::{tx::Msg, Any};
+use error_stack::ResultExt;
+use multiversx_sdk::data::address::Address;
+use serde::Deserialize;
+use tokio::sync::watch::Receiver;
+use tracing::info;
+
+use axelar_wasm_std::voting::{PollId, Vote};
+use events::Error::EventTypeMismatch;
+use events::Event;
+use events_derive::try_from;
+use voting_verifier::msg::ExecuteMsg;
+
+use crate::event_processor::EventHandler;
+use crate::handlers::errors::Error;
+use crate::mvx::proxy::MvxProxy;
+use crate::mvx::verifier::verify_message;
+use crate::types::{Hash, TMAddress};
+
+type Result<T> = error_stack::Result<T, Error>;
+
+#[derive(Deserialize, Debug)]
+pub struct Message {
+    pub tx_id: String,

```

Comment:
I tried multiple solutions here, but the problem is that the mvx library used to interact with the blockchain uses `&str` type for calls, to we need to go back to a string anyhow.
If we use `Hash`, then when we convert that to a string it adds the `0x` prefix which is not right, so I find this is the best way to do this with String to not complicate other parts of the code.

Reply:
Actually after testing on Devnet it seems that I can use Hash here, since the `tx_id` String here has the `0x` prefix as it comes from the Voting Verifier contract events.
Will have to implement a function which just gives me the hex string without `0x` prefix for Hash though so I can query the MultiversX blockchain properly.